### DISCUSSION TITLE
How can I add a button that toggles between scripts?

### DISCUSSION THREAD

**Discussion by gaveitatry2** [violentmonkey/violentmonkey#1804] [Q&A]
I have two user scripts. Let's just call them Script A and Script B. Script A makes the images on the webpage small and aligns them on the bottom right. Script B makes the images big so they are 100% width of the page.

Let's say that Script B is active. Normally, if I want to switch scripts, I click on the VioletMonkey icon and then I toggle Script B off and then toggle Script A on. And I also have to refresh the page after that.

But instead of doing that, I'd like to just have a button the webpage. If Script B is already active, pressing the button will cause Script B to toggle off, Script A to toggle on, and the website to reload automatically after that, all with just that one button press. And then if I press the button again, it will do everything except now it will switch back to Script A.

If that is possible, how is it done? I've only made simple scripts using simple stuff I found online that worked after trial and error.

**Comment by tophf** [comment]
Combine the code in a single script and toggle the functionality using an additional command under your script in the popup. The command is added by [GM_registerMenuCommand](https://violentmonkey.github.io/api/gm/#gm_registermenucommand):

```js
// @grant GM_getValue
// @grant GM_setValue
// @grant GM_registerMenuCommand
.....
let isBig = GM_getValue('big');
const getCmd = () => isBig ? 'Make small' : 'Make big';
GM_registerMenuCommand(getCmd(), function onCommand() {
  GM_unregisterMenuCommand(getCmd());
  isBig = !isBig; 
  GM_setValue('big', isBig); 
  GM_registerMenuCommand(getCmd(), onCommand);
  if (isBig) makeSmall(); else makeBig();
});

if (isBig) makeSmall(); else makeBig();
```

**Comment by gaveitatry2** [comment]
Thanks for the response. I didn't know where to put the two scripts. If you could edit your response with a PUT SCRIPT 1 HERE and a PUT SCRIPT 2 HERE, that would help. Also, I don't think that this puts a toggle button the webpage. It puts it in the browser toolbar VoiceMacro's popup menu. I wanted a button on the actual website that I could position using position:absolute; top: and left:. Also, will this refresh the page after it is toggled?
